sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala

```diff
@@ -1031,33 +1027,25 @@ private case class GpuParquetFileFilterHandler(
       case _ => false
     }
 
-  // TODO: After we deprecate Spark 3.1, fetch decimal meta with DecimalLogicalTypeAnnotation
-  @scala.annotation.nowarn("msg=method getDecimalMetadata in class PrimitiveType is deprecated")
-  private def canReadAsIntDecimal(pt: PrimitiveType, dt: DataType) = {
-    DecimalType.is32BitDecimalType(dt) && isDecimalTypeMatched(pt.getDecimalMetadata, dt)
-  }
-
-  // TODO: After we deprecate Spark 3.1, fetch decimal meta with DecimalLogicalTypeAnnotation
-  @scala.annotation.nowarn("msg=method getDecimalMetadata in class PrimitiveType is deprecated")
-  private def canReadAsLongDecimal(pt: PrimitiveType, dt: DataType): Boolean = {
-    DecimalType.is64BitDecimalType(dt) && isDecimalTypeMatched(pt.getDecimalMetadata, dt)
-  }
-
-  // TODO: After we deprecate Spark 3.1, fetch decimal meta with DecimalLogicalTypeAnnotation
+  // TODO: Since we have deprecated Spark 3.1, fetch decimal meta with DecimalLogicalTypeAnnotation
   @scala.annotation.nowarn("msg=method getDecimalMetadata in class PrimitiveType is deprecated")
-  private def canReadAsBinaryDecimal(pt: PrimitiveType, dt: DataType): Boolean = {
-    DecimalType.isByteArrayDecimalType(dt) && isDecimalTypeMatched(pt.getDecimalMetadata, dt)
+  private def canReadAsDecimal(pt: PrimitiveType, dt: DataType): Boolean = {
+    (DecimalType.is32BitDecimalType(dt)
+      || DecimalType.is64BitDecimalType(dt)
+      || DecimalType.isByteArrayDecimalType(dt)) && isDecimalTypeMatched(pt.getDecimalMetadata, dt)
   }
 
   // TODO: After we deprecate Spark 3.1, fetch decimal meta with DecimalLogicalTypeAnnotation
   @scala.annotation.nowarn("msg=class DecimalMetadata in package schema is deprecated")
   private def isDecimalTypeMatched(metadata: DecimalMetadata,
```

Comment:
I would have named this `isDecimalTypeCompatible`, but it looks like Apache Spark names it this way as well.